tokio/tests/task_join_set.rs

```diff
@@ -227,3 +227,80 @@ async fn join_set_coop() {
     assert!(coop_count >= 1);
     assert_eq!(count, TASK_NUM);
 }
+
+#[tokio::test]
+async fn try_join_next() {
+    const TASK_NUM: u32 = 1000;
+
+    let (send, recv) = tokio::sync::watch::channel(());
+
+    let mut set = JoinSet::new();
+
+    for _ in 0..TASK_NUM {
+        let mut recv = recv.clone();
+        set.spawn(async move { recv.changed().await.unwrap() });
+    }
+    drop(recv);
+
+    assert!(set.try_join_next().is_none());
+
+    send.send_replace(());
+    send.closed().await;
```

Comment:
It's not actually guaranteed that the tasks are considered finished by the `JoinSet` when `closed().await` returns. This is because tasks are marked finished _after_ the destructor runs, but the channels are dropped while the destructor runs.

If you make this into a current-thread runtime, then this race goes away.

Reply:
Ah you're right. Fixed it.